driver-core/src/main/java/com/datastax/driver/core/RequestHandler.java

```diff
@@ -694,11 +722,20 @@ public void onException(Connection connection, Exception exception, long latency
                 connection.release();
 
                 if (exception instanceof ConnectionException) {
-                    if (metricsEnabled())
+                    RetryPolicy retryPolicy = retryPolicy();
+                    RetryPolicy.RetryDecision decision;
+                    if (retryPolicy instanceof ExtendedRetryPolicy)
+                        decision = ((ExtendedRetryPolicy)retryPolicy).onUnexpectedError(statement, request().consistency(), retriesByPolicy, requestSent);
+                    else
+                        decision = RetryPolicy.RetryDecision.tryNextHost(null);
```

Comment:
The documentation for RetryDecision#getRetryConsistencyLevel indicates that it returns null for RETRY or RETHROW cases.  We should update that doc as it now seems stale as in the case retry CL is null, it simply reuses the statement.